test/features/schema_tests.js

```diff
@@ -101,3 +102,30 @@ parallel('Responses should conform to the provided JSON schema of the response',
         });
     });
 });
+
+describe('Swagger Spec', () => {
+    const runner  = new ServiceRunner();
+    const config = server.loadConfig(`${__dirname}/../../config.example.wikimedia.yaml`);
+    config.logging.level = "error";
+    let swagger;
+
+    before(() => server.stop());
+
+    after(() => runner.stop());
+
+    it('should be compliant with Swagger specifications', () => {
+        return runner.start(config)
+            .then(() => { return preq.get({ uri: `${server.config.baseURL}/?spec` }); })
+            .then((res) => {
+                swagger = JSON.stringify(res.body);
+            }).then(() => {
+                return preq.post({uri: `http://online.swagger.io/validator/debug`, body: swagger})
+                    .then((res) => {
+                        // if valid will return an empty object: {}
+                        if (Object.keys(res.body).length !== 0) {
+                            throw new Error(`Swagger specification is not valid: ${res.body}`)
+                        }
+                    })
+            })
+    });
+});
```

Comment:
This has to be rebased after #1101 is merged and the spec validity test has to be moved here https://github.com/wikimedia/restbase/pull/1101/files#diff-ff16a5ee31f1814844151cee8c98d8aaR38